### ISSUE TITLE
Calltrace does not continue after first time hitting `break`

### ISSUE THREAD

**Issue by sebglatz** [Auterion/embedded-debug-tools#4]
# TL;DR
`px4_breaktrace` does not continue to run after `break` despite `continue`
The MCU halts forever and thus only one function call is sampled.

# What?
I want to create a callgraph for some SPI function. To do so, I run:
`python3 -m emdbg.analyze.calltrace -v --px4-dir ~/PX4-Autopilot --target px4_fmu-v5_default --jlink --sample 60 -ex "px4_calltrace_spi_exchange"

with `px4_calltrace_spi_exchange` defined like this in `fmu.gdb`:
```
define px4_calltrace_spi_exchange
    px4_breaktrace spi_exchange
end
```
# Expected behavior
The MCU should continue to run after hitting the break point.
Note: The function `px4_commands_backtrace` does call `continue` after `break` and some `printf`

**Comment by niklaut** [comment]
Hm, I cannot reproduce this:

```
python3 -m emdbg.analyze.calltrace -v --target px4_fmu-v5x_default --stlink --sample 60 -ex "px4_calltrace_spi_exchange"
python3 -m emdbg.analyze.calltrace -v --target px4_fmu-v5x_default --stlink --sample 60 -ex "break spi_exchange"
```

![calltrace_breakspi_exchange](https://github.com/Auterion/embedded-debug-tools/assets/121870655/d2c3802d-89cd-4bb9-916d-5c5c236a50c7)

![calltrace_px4_calltrace_spi_exchange](https://github.com/Auterion/embedded-debug-tools/assets/121870655/e1de4d47-7b79-4cc8-834c-60a6f975b8c4)


Perhaps it's because I'm using an STLink, let me try with a JLink.

**Comment by sebglatz** [comment]
Thank you.
I have just tried it with an stlink v2. 
The MCU also halts forever.

**Comment by niklaut** [comment]
Hm, that sucks. What GDB version are you using (`arm-none-eabi-gdb-py3 --version`)? Is the ELF file in sync with the firmware? ie. try
```sh
python3 -m emdbg.analyze.calltrace -v --target px4_fmu-v5x_default --stlink --sample 60 -ex "load" -ex "px4_reset" -ex "break spi_exchange"
```

**Comment by niklaut** [comment]
Oh I thought it was a typo at first. I'll check if I can find a v5 somewhere. But at first glance it looks pretty similar to the v5x.

- Can you use GDB to put a breakpoint on `spi_exchange` and then continue?
- You can reduce the SWD clock speed in `fmu_v5x_stlink.cfg` from 8MHz down to maybe 1MHz and see if it works?

**Comment by sebglatz** [comment]
- Yes, that's so confusing to me: `break spi_exchange` and then executing the commands of `px4_commands_backtrace`  "by hand" (`printf...`, `px4_backtrace`, `continue`) during a gdb session works nicely with the jlink (have to try stlink, but prob. same).

- Reducing the stlink speed to 1MHz did not help, unfortunately...

**Comment by niklaut** [comment]
Hm, maybe it's a dependency problem? Here's what I have installed:

```
$ pip3.8 install --upgrade-strategy=eager emdbg
Requirement already satisfied: emdbg in /opt/homebrew/lib/python3.8/site-packages (1.3.9.post2+git.4f062ccb.dirty)
Requirement already satisfied: pyserial in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (3.5)
Requirement already satisfied: graphviz in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (0.20.1)
Requirement already satisfied: pygdbmi in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (0.11.0.0)
Requirement already satisfied: cmsis-svd in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (0.4)
Requirement already satisfied: arm-gdb in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (0.9.8)
Requirement already satisfied: pydwf in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (1.1.13)
Requirement already satisfied: yoctopuce in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (1.10.54852)
Requirement already satisfied: rpyc in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (5.3.1)
Requirement already satisfied: rich in /opt/homebrew/lib/python3.8/site-packages (from emdbg) (13.6.0)
Requirement already satisfied: six>=1.10 in /opt/homebrew/lib/python3.8/site-packages (from cmsis-svd->emdbg) (1.16.0)
Requirement already satisfied: numpy in /opt/homebrew/lib/python3.8/site-packages (from pydwf->emdbg) (1.24.4)
Requirement already satisfied: markdown-it-py>=2.2.0 in /opt/homebrew/lib/python3.8/site-packages (from rich->emdbg) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /opt/homebrew/lib/python3.8/site-packages (from rich->emdbg) (2.15.1)
Requirement already satisfied: typing-extensions<5.0,>=4.0.0 in /opt/homebrew/lib/python3.8/site-packages (from rich->emdbg) (4.8.0)
Requirement already satisfied: plumbum in /opt/homebrew/lib/python3.8/site-packages (from rpyc->emdbg) (1.8.2)
Requirement already satisfied: mdurl~=0.1 in /opt/homebrew/lib/python3.8/site-packages (from markdown-it-py>=2.2.0->rich->emdbg) (0.1.2)
```

**Comment by sebglatz** [comment]
Thank you.
Exact same dependencies. (except emdbg which is 1.3.9, not the dirty hash)
Also set it up freshly in a virtual env (python 3.8.18).
Still not sampling more than once...

**Comment by niklaut** [comment]
I'm not sure what else could be the problem… did this magically solve itself?
